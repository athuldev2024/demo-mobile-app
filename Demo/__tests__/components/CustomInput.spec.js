import React from 'react';
import {render} from '@testing-library/react-native';
import CustomInput from '../../src/components/common/CustomInput';
import {PaperProvider} from 'react-native-paper';

jest.mock('react-native-paper', () => {
  const {TextInput, View, Text} = require('react-native');

  return {
    PaperProvider: ({children}) => <View>{children}</View>,
    TextInput: jest.fn(
      ({label, placeholder, value, onChangeText, editable}) => (
        <View>
          {label && <Text>{label}</Text>}
          <TextInput
            placeholder={placeholder}
            value={value}
            onChangeText={onChangeText}
            editable={editable}
          />
        </View>
      ),
    ),
  };
});

describe('CustomInput Component', () => {
  const onChangeTextMock = jest.fn();

  const renderWithPaperProvider = (props = {}) => {
    const defaultProps = {
      value: '',
      onChangeText: onChangeTextMock,
      placeholder: 'sample placeholder',
      keyboardType: 'default',
      secureTextEntry: false,
      touched: {},
      errors: {},
      fieldName: 'name',
    };

    return render(
      <PaperProvider>
        <CustomInput {...defaultProps} {...props} />
      </PaperProvider>,
    );
  };

  it('renders the TextInput and check if it exists', () => {
    const {getByPlaceholderText} = renderWithPaperProvider();
    expect(getByPlaceholderText('sample placeholder')).toBeTruthy();
  });

  it('renders the TextInput awith error message', () => {
    const {getByText} = renderWithPaperProvider({
      touched: {name: 'touched'},
      errors: {name: 'name error message'},
      fieldName: 'name',
    });
    expect(getByText('name error message')).toBeTruthy();
  });
});
