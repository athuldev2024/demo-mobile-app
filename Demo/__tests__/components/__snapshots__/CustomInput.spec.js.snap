// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomInput Component matches snapshot and renders properly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={{}}
    >
      <View
        pointerEvents="none"
        style={
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 6,
            },
            false,
            {
              "backgroundColor": "rgba(255, 251, 254, 1)",
              "borderColor": "rgba(121, 116, 126, 1)",
              "borderRadius": 4,
              "borderWidth": 1,
            },
            undefined,
          ]
        }
        testID="text-input-outline"
      />
      <View
        style={
          [
            {
              "flexGrow": 1,
              "paddingBottom": 0,
            },
            {
              "minHeight": 56,
              "paddingTop": 8,
            },
          ]
        }
      >
        <View
          pointerEvents="none"
          style={
            [
              {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              {
                "overflow": "hidden",
              },
            ]
          }
        >
          <View
            collapsable={false}
            pointerEvents="none"
            style={
              {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": [
                  {
                    "translateX": 0,
                  },
                ],
                "width": 750,
                "zIndex": 3,
              }
            }
          >
            <View
              style={
                {
                  "width": 73,
                }
              }
            >
              <Text
                collapsable={false}
                maxFontSizeMultiplier={1.5}
                numberOfLines={1}
                style={
                  {
                    "backgroundColor": "rgba(255, 251, 254, 1)",
                    "color": "transparent",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "height": 0,
                    "left": 8,
                    "letterSpacing": 0.15,
                    "lineHeight": undefined,
                    "opacity": -0.6666666666666667,
                    "paddingHorizontal": 0,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 33,
                    "transform": [
                      {
                        "translateX": 0,
                      },
                      {
                        "translateY": 0,
                      },
                      {
                        "scale": 1,
                      },
                      {
                        "scaleY": 1,
                      },
                    ],
                    "width": -16,
                    "writingDirection": "ltr",
                  }
                }
                testID="text-input-outlined-label-background"
              />
              <Text
                collapsable={false}
                maxFontSizeMultiplier={1.5}
                numberOfLines={1}
                onLayout={[Function]}
                onTextLayout={[Function]}
                style={
                  {
                    "color": "rgba(103, 80, 164, 1)",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 0,
                    "letterSpacing": 0.15,
                    "lineHeight": undefined,
                    "maxWidth": 73,
                    "opacity": 0,
                    "paddingHorizontal": 16,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 32,
                    "transform": [
                      {
                        "translateX": 0,
                      },
                      {
                        "translateY": 0,
                      },
                      {
                        "scale": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
                testID="text-input-outlined-label-active"
              >
                Enter your name
              </Text>
              <Text
                collapsable={false}
                maxFontSizeMultiplier={1.5}
                numberOfLines={1}
                style={
                  {
                    "color": "rgba(73, 69, 79, 1)",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 0,
                    "letterSpacing": 0.15,
                    "lineHeight": undefined,
                    "maxWidth": 73,
                    "opacity": 0,
                    "paddingHorizontal": 16,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 32,
                    "transform": [
                      {
                        "translateX": 0,
                      },
                      {
                        "translateY": 0,
                      },
                      {
                        "scale": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
                testID="text-input-outlined-label-inactive"
              >
                Enter your name
              </Text>
            </View>
          </View>
        </View>
        <TextInput
          cursorColor="rgba(103, 80, 164, 1)"
          editable={true}
          maxFontSizeMultiplier={1.5}
          multiline={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          onLayout={[Function]}
          placeholder=" "
          placeholderTextColor="rgba(73, 69, 79, 1)"
          selectionColor="rgba(103, 80, 164, 1)"
          style={
            [
              {
                "flexGrow": 1,
                "margin": 0,
              },
              {
                "height": 48,
              },
              {
                "paddingBottom": 0,
                "paddingTop": 0,
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "letterSpacing": 0.15,
                "lineHeight": undefined,
                "minWidth": 65,
                "paddingHorizontal": 16,
                "textAlign": "left",
                "textAlignVertical": "center",
              },
              false,
              [
                {},
              ],
              undefined,
            ]
          }
          testID="text-input-outlined"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
